src/Nethermind/Nethermind.Evm/CodeInfoRepository.cs

```diff
@@ -169,11 +153,140 @@ public void InsertCode(IWorldState state, ReadOnlyMemory<byte> code, Address cod
         _codeCache.Set(codeHash, codeInfo);
     }
 
+    /// <summary>
+    /// Insert code delegations from transaction authorization_list authorized by signature,
+    /// and return all authority addresses that was accessed and amount of autorization refunds.
+    /// eip-7702
+    /// </summary>
+    public int InsertFromAuthorizations(
+        IWorldState worldState,
+        AuthorizationTuple?[] authorizations,
+        ISet<Address> accessedAddresses,
+        IReleaseSpec spec)
+    {
+        int refunds = 0;
+        //TODO optimize
+        foreach (AuthorizationTuple? authTuple in authorizations)
+        {
+            if (authTuple is null)
+                continue;
+            authTuple.Authority = authTuple.Authority ?? _ethereumEcdsa.RecoverAddress(authTuple);
+            string? error;
+
+            if (!IsValidForExecution(authTuple, worldState, _ethereumEcdsa.ChainId, accessedAddresses, out error))
+                continue;
+
+            if (worldState.AccountExists(authTuple.Authority))
+                refunds++;
+            else
+                worldState.CreateAccount(authTuple.Authority, 0);
+
+            InsertDelegationCode(worldState, authTuple.CodeAddress, authTuple.Authority, spec);
+            worldState.IncrementNonce(authTuple.Authority);
```

Comment:
Changed so it will create a new account with nonce set to 1, if account doesn't exists. 

I tried to benchmark a parallel version and it performed way worse, mainly because of use of `Interlocked.Increment`.
Additionally it doesn't look like state access is thread safe.

Reply:
Do you have more ideas for optimization here? @MarekM25